### PR TITLE
Working routing covers all states

### PR DESCRIPTION
**Pull request by sashkivna**
version with deleted comments and +- beautified code

### CODE REVIEW

File: app/components/todo/todo.directive.js [hunk 1]
```diff
@@ -22,21 +23,15 @@ angular.module('todo.module')
                             + currentdate.getHours() + ":"
                             + currentdate.getMinutes() + ":"
                             + currentdate.getSeconds();
-                        $scope.tasks.push({task: $scope.title, cteatedAt: datetime, /*user: currentUser,*/ checked: false});
+                        $scope.tasks.push({task: $scope.title, cteated: datetime, done: false});
+                        console.log($scope.tasks);
```
Comment (r0ss0neri) [inline app/components/todo/todo.directive.js]: remove `console.log` statements if they are not needed

File: app/components/todo/todo.directive.js [hunk 2]
```diff
@@ -8,9 +8,10 @@ angular.module('todo.module')
                 tasks: '='
             },
 
-            link: function ($scope) {
-                    $scope.tasks = [];
+            controller: function () {
```
Comment (r0ss0neri) [inline app/components/todo/todo.directive.js]: remove controller

File: app/components/todo/todo.directive.js [hunk 3]
```diff
@@ -22,21 +23,15 @@ angular.module('todo.module')
                             + currentdate.getHours() + ":"
                             + currentdate.getMinutes() + ":"
                             + currentdate.getSeconds();
-                        $scope.tasks.push({task: $scope.title, cteatedAt: datetime, /*user: currentUser,*/ checked: false});
+                        $scope.tasks.push({task: $scope.title, cteated: datetime, done: false});
```
Comment (r0ss0neri) [inline app/components/todo/todo.directive.js]: format 
    
    $scope.tasks.push({
        task: $scope.title, 
        created: datetime, 
        done: false
    });

File: app/components/todo/todo.service.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
+angular.module('app.module')
+    .factory('todoService', function (userService) {
+        var users = JSON.parse(localStorage.getItem('users')) || [],
+            currentUser = null;
+
+        function updateTasks(tasks) {
+            cauurentUser = userService.getCurrentUser();
```
Comment (r0ss0neri) [inline app/components/todo/todo.service.js]: blank line after

File: app/components/todo/todo.service.js [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+            if (tasks.length !== 0) {
+                for (var i = 0; i < tasks.length; i++) {
+                    if (!tasks[i].email) {
+                        tasks[i].email = cauurentUser.email;
+                    }
+                }
+            }
```
Comment (r0ss0neri) [inline app/components/todo/todo.service.js]: blank line after

File: app/components/todo/todo.service.js [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+            localStorage.setItem('tasks', JSON.stringify(tasks));
+
+        }
+
+        return {
+            users: users,
```
Comment (r0ss0neri) [inline app/components/todo/todo.service.js:18]: `users ` -> `getAllUserDescriptors`

File: app/login/register-controller.js [hunk 7]
```diff
@@ -1,5 +1,5 @@
 angular.module('app.module')
-    .controller('loginController', function ($scope, userService) {
+    .controller('loginController', function ($scope, userService, $state) {
```
Comment (r0ss0neri) [inline app/login/register-controller.js]: don't forget to sort injectors alphabetically 

- angular services
- then third party libraries 
- app dependencies

File: app/main/main.controller.js [hunk 8]
```diff
@@ -0,0 +1,8 @@
+angular.module('app.module')
+    .controller('mainController', function ($scope, todoService) {
+        $scope.tasks = JSON.parse(localStorage.getItem('tasks')) || [];
```
Comment (r0ss0neri) [inline app/main/main.controller.js]: can we use todoService to get tasks instead of localStorage?

File: app/users-todo/users-todo.controller.js [hunk 9]
```diff
@@ -1,8 +1,9 @@
 angular.module('app.module')
-    .controller('usersCntr', function ($rootScope, $scope, appCasheService, $state) {
-        $scope.usersFromCashe = appCasheService.getUsersFromCashe();
+    .controller('usersCntr', function ($scope, $state) {
+        $scope.users = JSON.parse(localStorage.getItem('users')) || [];
 
         $scope.showTodo = function () {
+            debugger;
```
Comment (r0ss0neri) [inline app/users-todo/users-todo.controller.js]: remove debugger

**Review by r0ss0neri — CHANGES_REQUESTED** [review]

File: app/components/todo/todo.directive.js (hunk 1 above)
Comment (sashkivna) [reply app/components/todo/todo.directive.js]: done

File: app/components/todo/todo.directive.js (hunk 2 above)
Comment (sashkivna) [reply app/components/todo/todo.directive.js]: done

File: app/components/todo/todo.directive.js (hunk 3 above)
Comment (sashkivna) [reply app/components/todo/todo.directive.js]: done

File: app/components/todo/todo.service.js (hunk 4 above)
Comment (sashkivna) [reply app/components/todo/todo.service.js]: done

File: app/login/register-controller.js (hunk 7 above)
Comment (sashkivna) [reply app/login/register-controller.js]: changed

File: app/users-todo/users-todo.controller.js (hunk 9 above)
Comment (sashkivna) [reply app/users-todo/users-todo.controller.js]: deleted

File: app/main/main.controller.js (hunk 8 above)
Comment (sashkivna) [reply app/main/main.controller.js]: changed

File: app/users-todo/users-todo.controller.js [hunk 10]
```diff
@@ -1,8 +1,13 @@
 angular.module('app.module')
-    .controller('usersCntr', function ($rootScope, $scope, appCasheService, $state) {
-        $scope.usersFromCashe = appCasheService.getUsersFromCashe();
+    .controller('usersCntr', function ($scope, $state, userService) {
+        $scope.users = JSON.parse(localStorage.getItem('users')) || [];
```
Comment (r0ss0neri) [inline app/users-todo/users-todo.controller.js]: use `userService` to get all users

**Review by r0ss0neri — CHANGES_REQUESTED** [review]

File: app/users-todo/users-todo.controller.js (hunk 10 above)
Comment (sashkivna) [reply app/users-todo/users-todo.controller.js]: done

**Pull request merged**